File: 118.Pascal's Triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        res = []
        for i in range(numRows):
            resl = []
            for j in range(i+1):
                if j == 0:
                    resl.append(1);
                elif j == i:
                    resl.append(1);
                else:
                    resl.append(res[i-1][j-1]+res[i-1][j])
            res.append(resl);  
        return res
        

File: 118.Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        for(int i=0; i<numRows; i++){
            vector<int> resl;
            for(int j=0; j<=i; j++){
                if(j==0 || j==i)
                    resl.push_back(1);
                else
                    resl.push_back(res[i-1][j-1]+res[i-1][j]);
            }
            res.push_back(resl);
        }
        return res;
    }
};
